### PR TITLE
Return loss, but not predictions and target during training steps to save GPU memory

### PR DESCRIPTION
**Pull request by crlna16**
Note that at each step, besides the loss, predictions and target are generated and put in the return dictionary. This dictionary could be used at training_epoch_end , but the values are never required. Still they are kept in GPU memory, which is then growing during training up to a point where it can exceed available GPU memory (as happened when we were using large images in the validation set). The proposed solution is to return only the loss by default, as the error was hard to find when it occurs, 

More notes on our debugging process:

https://medium.com/@caroline.arnold_63207/reclaim-your-gpu-memory-5fb80a980a09

### CODE REVIEW

**Review by IsraMekki0 — APPROVED** [review]
Great! Thanks Caroline!
